include/swift/Reflection/Records.h

```diff
@@ -374,6 +374,22 @@ class BuiltinTypeDescriptor {
   }
 };
 
+class MultiPayloadEnumDescriptor {
+public:
+  const RelativeDirectPointer<const char> TypeName;
+  uint32_t Flags;
+  uint32_t BitfieldBitCount; // Size of bitfield in bits
+  uint8_t  BitfieldBits[]; // Variably-sized bitmask field; always a multiple of 4 bytes
```

Comment:
TODO: This descriptor is inherently variably-sized.  Should I include an explicit `size` field so it can potentially be expanded in the future?

Reply:
@mikeash Should I worry about evolution for reflection records?  It doesn't look like our existing reflection records have any support for being expanded in the future.  This seems potentially problematic, though maybe it's okay to "just" add a new section every time we discover we need a new field?

Reply:
I worked up an architecture here that allows this record to add new properties in the future:  There's a single `uint32_t contents[]` array that holds all data, and standard boilerplate to compute the index of any particular field stored in that array.  I've included a commented-out template showing how to add a new property.